### PR TITLE
Add TA troubleshooting page

### PR DESCRIPTION
**Pull request by avillela**
Adding page to the docs with tips for troubleshooting the OTel Operator's Target Allocator.

This was [originally going to be a blog post](#4700); however, per suggestion from @svrnm, this is a much better fit.

Reference issue: #4707

### CODE REVIEW

**Comment by avillela** [comment]
Question for @open-telemetry/operator-approvers - I have a [working example repo](https://github.com/avillela/otel-target-allocator-talk) that I'd like to reference on this page, but I think that it's probably best to move this code to an examples folder the OTel Operator repo before doing this. Thoughts?

File: content/en/docs/kubernetes/operator/troubleshooting/_index.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/_index.md:12]:
```suggestion
```

Remove that, those redirects should only be introduced if pages existed previously and have been moved

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:3]:
```suggestion
```

The description should either provide some additional details to the title or skip it

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:16]:
```suggestion
### Did you deploy all of your resources to Kubernetes?
```

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:22]:
```suggestion
### Do you know if metrics are actually being scraped?
```

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:26]: Something seems off with this link. Also is there a target to an official prometheus documentation?

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:86]: We might consider auto-updating this collector version here, but this can happen in a later update

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:268]: Can we move that to docs as well? Also can we point to HEAD of main instead of a specific commit?

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:272]:
```suggestion
### Is the Target Allocator enabled? Is Prometheus service discovery enabled?
```

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:306]:
```suggestion
### Did you configure a ServiceMonitor (or PodMonitor) selector?
```

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:370]:
```suggestion
### Did you leave out the serviceMonitorSelector and/or podMonitorSelector configuration altogether?
```

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:373]:
```suggestion
["Did you configure a ServiceMonitor or PodMonitor selector"](#4--did-you-configure-a-servicemonitor-or-podmonitor-selector), setting
```

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:398]:
```suggestion
### Do your labels, namespaces, and ports match for your ServiceMonitor and your Service (or PodMonitor and your Pod)?
```

**Review by svrnm — DISMISSED** [review]
Thanks, this is a really good addition to docs! A few comments and suggestions. Note that the section titles are really long which might look strange in the outline navigation, maybe we can review and shorten them

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:85]: @svrnm I can also leave it out, and then the OTel Operator will just use whatever the default image is

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md]: @svrnm I've updated the link to point to main instead of the specific commit. Just to confirm - this links to the Operator's Target Allocator README. Would we want to move all contents from this readme to the docs and make that the single source of truth?

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md]: Done!

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (svrnm) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:85]: if that works, I would prefer that, except someone from @open-telemetry/operator-approvers disagrees

Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:85]: I'll hold off on updating until we get feedback from @open-telemetry/operator-approvers, then. 🙂

**Comment by avillela** [comment]
> Thanks, this is a really good addition to docs! A few comments and suggestions. Note that the section titles are really long which might look strange in the outline navigation, maybe we can review and shorten them

Yeah, they do look a bit long. I think that when we had the step numbers, it was a bit easier to distinguish one from the other. Is there a reson why you prefer to leave those out, @svrnm?

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (theletterf) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:439]:
```suggestion
For example, the following `Service` resource would get picked up by the
`ServiceMonitor`, because it matches the previous criteria:
```

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (theletterf) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:477]: Some humor in docs — thanks! :D

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (theletterf) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:484]:
```suggestion
Kubernetes pods that match on labels, namespaces, and named ports.
```

**Review by theletterf — CHANGES_REQUESTED** [review]
Good stuff! Added some comments.

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (swiatekm-sumo) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:85]: It's better to leave it blank in examples, as a rule. We do run the risk of our examples being broken by future releases, but in the case of K8s API Versioning, breaking changes can only happen with an explicit version change, the worst that can happen is that the whole resource will be rejected as unsupported. There won't be subtle breakage.

Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md]: Noted, and removed!

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md]: Done!

**Comment by avillela** [comment]
FYI @svrnm @swiatekm-sumo @theletterf - I had some previous content for troubleshooting auto-instrumentation, so I creaded a page for that too.

**Comment by svrnm** [comment]
> FYI @svrnm @swiatekm-sumo @theletterf - I had some previous content for troubleshooting auto-instrumentation, so I creaded a page for that too.

Can you fork that out in a dedicated PR or is this closely related?

**Review by theletterf — APPROVED** [review]

**Comment by avillela** [comment]
> > FYI @svrnm @swiatekm-sumo @theletterf - I had some previous content for troubleshooting auto-instrumentation, so I creaded a page for that too.
> 
> Can you fork that out in a dedicated PR or is this closely related?

It's not closely related, so I can definitely create a separate PR.

**Comment by avillela** [comment]
Removed auto-instrumentation troubleshooting page. I have created a separate PR for this (#4724).

File: content/en/docs/kubernetes/operator/troubleshooting/_index.md
Comment (swiatekm-sumo) [inline content/en/docs/kubernetes/operator/troubleshooting/_index.md:7]: As a point of clarification, the target allocator doesn't *scrape metrics*. It *discovers scrape targets* and *allocates them to collectors*. This misunderstanding appears in other places in this PR as well, and it should be corrected there as well.

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (swiatekm-sumo) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:185]:
```suggestion
see what scrape targets are getting picked up by running `curl` against the value of
```

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (swiatekm-sumo) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:22]:
```suggestion
Target Allocator is discovering scrape targets from your
```

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (swiatekm-sumo) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:329]: This would be correct in apiVersion `v1alpha1`, but in `v1beta1` this field is a label selector, and should look like this:

```suggestion
      serviceMonitorSelector:
        matchLabels:
          app: my-app
```

You also have the documentation for v1alpha1 linked above, which is most likely the source of the confusion here. The v1beta1 struct is documented [here](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr-1).

Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:329]: Oooh, thanks for that catch!

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:309]:
```suggestion
[`serviceMonitorSelector`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr-1).
```

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:312]:
```suggestion
[`serviceMonitorSelector`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr-1)
```

File: content/en/docs/kubernetes/operator/troubleshooting/_index.md
Comment (avillela) [inline content/en/docs/kubernetes/operator/troubleshooting/_index.md:7]:
```suggestion
  allocator is failing to discover scrape targets for allocation to collectors.
```

**Review by avillela — COMMENTED** [review]
Applying my own updates per feedback below

**Comment by avillela** [comment]
@swiatekm-sumo thanks for keeping me honest! I've incorporated your suggestions and did another read through. Hopefully I didn't miss any spots where I mis-explained the TA functionality.

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (jaronoff97) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:138]: we could also give people the query for the service by doing `kubectl get svc -l app.kubernetes.io/component=opentelemetry-targetallocator`

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (jaronoff97) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:180]:
```suggestion
This is good news, because it tells us that the scrape config discovery is working!
```

Service discovery working is the next step where we check that targets are showing up for these scrape_configs' service discovery.

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (jaronoff97) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:183]: this entry is because [the config provided](https://github.com/open-telemetry/opentelemetry.io/pull/4708/files#diff-1dbe631d16dc9eb3b4e3af1e41c754015d43209646e535c073e124e5c4a7e2e9R100-R103) has the otel-collector self scrape set, the TA wouldn't have it if not for that block

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (jaronoff97) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:190]: this is great to see – i love showing the full curl and sample output for what this should look like! one thing... this curl's URL isn't found in the previous output... I think it should be `/jobs/serviceMonitor%2Fopentelemetry%2Fsm-example%2F0/targets`

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (jaronoff97) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:198]: would be good to note here that the query parameter collector_id is saying that these are the targets for otelcol-collector-0

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (jaronoff97) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:387]: i would include that setting it like this means it will match everything.

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (jaronoff97) [inline content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:463]: This is such a great example and _exactly_ what I helped a user debug in slack a few weeks ago! Thanks for including it.

**Review by jaronoff97 — COMMENTED** [review]
A few minor comments, but overall this is fantastic and will be massively helpful!

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:138]: Thanks! Added that.

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:183]: Thanks for clarifying! Added.

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:190]: Good catch! Fixed.

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:198]: Noted!

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:387]: Done!

File: content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md
Comment (avillela) [reply content/en/docs/kubernetes/operator/troubleshooting/target-allocator.md:463]: ❤️

**Review by jaronoff97 — APPROVED** [review]

**Comment by jaronoff97** [comment]
THANK YOU @avillela fantastic work here :D :D :D

**Review by jaronoff97 — APPROVED** [review]

**Review by swiatekm-sumo — APPROVED** [review]
Nice 💪

**Review by cartermp — APPROVED** [review]

**Pull request merged**
